rotkehlchen/chain/evm/decoding/decoder.py

```diff
@@ -304,14 +304,15 @@ def decode_transaction(
                 address=tx_log.address,
                 chain_id=self.evm_inquirer.chain_id,
             )
-            event = self.try_all_rules(
+            event, new_action_items = self.try_all_rules(
                 token=token,
                 tx_log=tx_log,
                 transaction=transaction,
                 decoded_events=events,
                 action_items=action_items,
                 all_logs=tx_receipt.logs,
             )
+            action_items.extend(new_action_items)
             if event:
```

Comment:
is something that should be changed but can be done if left has more comments. This should be explicit check 

```suggestion
            action_items.extend(new_action_items)
            if event is not None:
```

Reply:
+1, please do this